doorstop/core/tree.py

```diff
@@ -176,9 +176,13 @@ def _place(self, document):
             log.info("parent options: {}".format(prefixes))
             raise DoorstopError(msg)
 
+        self.regenerate_children()
+
+    def regenerate_children(self):
```

Comment:
It was a name clash between the 'document' argument to _place and the 'document' used in the for loop. I couldn't think of a better name for the second use, so I put it in a separate scope. You're right, it should be a private method. I have raised #251 which removes that patch and I'll have another look at this issue tomorrow.